api/src/main/java/org/eclipse/microprofile/lra/client/LRAClient.java

```diff
@@ -188,120 +185,20 @@ URL startLRA(String clientID, Long timeout, ChronoUnit unit,
      *
      * @param lraId The unique identifier of the LRA (required)
      *
-     * {@link CompensatorStatus#name()}. If the final status is not returned the
-     *             client can still discover the final state using the
-     *             {@link LRAClient#getStatus(URL)} method
-     *
      * @throws NotFoundException if the LRA no longer exists
      *
      * @throws GenericLRAException Communication error (the reason is availalbe via
      * the {@link GenericLRAException#getStatusCode()} method
      *
      * @return the response MAY contain the final status of the LRA as reported by
+     *         {@link LRAStatus#name()}
+     *         If the final status is not returned the
+     *         client can only discover the final state using proprietary APIs
```

Comment:
@tomjenkinson This is where the problem with moving the query API into the TCK lies

Reply:
Can you expand on that a little please? I imagine proprietary is OK because that would be the same for JTA JTS etc right?

Reply:
So the caller uses the client API to close an LRA and gets back something like status `Closing` (since some participants may still be completing). But now the caller can never discover the final outcome of his request to close the LRA since there is no interface for that.

Reply:
Rudy, there is also some discussion about this in the issue #19 where I accepted Tom's point of view.

Reply:
I imagine vendors could provide some method to determine the outcome of a call that returned "Closing" - so that situation is fine fom spec point of view IMO

Reply:
I think we should not be encouraging usage of proprietary APIs. At least as I understand it, MicroProfile aims to have an easily replaceable implementations without the need of code changes in the user code. So users should be able to get all required information through LRAClient API. For this use-case maybe we can instead return Future<LRAStatus> or make the call blocking until definite result can be provided?

Reply:
I like the idea of the future for normal use case.

There would still be some need for a proprietary or otherwise way to deal with a client application that crashed after issuing the complete but before being able to obtain the result.

Reply:
Returning a Future is going to be difficult to implement in all cases as it needs to remain valid potentially forever. The `Closing` state is provided to indicate that the request is in progress which therefor obviates the need for Futures.

Reply:
Couldn;t the future return an exception (NoLongerAvailable) if it is polled at some point in violation of the protocol?

Reply:
I mean that it is valid for a participant to take an arbitrarily long period to complete.

There is also the issue of the client failing which would mean he does not have access to the future when he restarts.

And I still stand by my point that if we block or return a future from the close call then the Completing states become redundant.

Reply:
both valid points. I would also be more inclined to keep single `getStatus(lraId)` method. Can someone reiterate why we should remove it please?

Reply:
I was thinking some implementations might not be able to provide it immediately but perhaps "Try again later" can be a return code so long as it is certain at some point it could provide it.

I am considering "routing slip" or "event sourcing" potential implementations

Reply:
I think both of these concepts are capable of providing LRA status information. In my view LRAClient#getStatus provides a client view of the status of the LRA (note that saga is defined as eventually consistent) so I don't see any problem if for instance routing slip will be returning Cancelling/Completing until the slip is again received by this particular service.

Reply:
what status could it return if it was disconnected from the slip and recovered with a reference to the LRA ID though, it would not know if it was cancelling or completing. I think something like "Try later" might be needed for that scenario

Reply:
maybe 503

Reply:
Understood, but I would still stand by the fact that client.getStatus() provides a client view of the LRA. If it doesn't know, it can say that (e.g. again by exception). In orchestration, if both client and coordinator fail and client recovers first with the lraId but the coordinator is still down we would have similar situation right?

Reply:
yes, we would

Reply:
How are we going to run the TCK against an implementation that does not contain an API for reporting the status of an LRA?

Reply:
we will definitely need such method. I believe the question discussed is whether it will be in the spec API or in the TCK SPI.

Reply:
It is the question

Reply:
If the implementation can provide one for the TCK why can't it also provide one for the Client API too?